### ISSUE TITLE
parse bibtex entries by any field

### ISSUE THREAD

**Issue by asetGem** [texstudio-org/texstudio#1762]
#### Current behaviour: 

When typing `\cite{` texstudio shows a popup with a list of bibtex entries, and we can scroll up/down to find the one needed. We can also type the first letter(s) of the bibtex entry to be more specific (it filters the entries to show only the corresponding ones).

#### Feature:

1) When typing the first letter(s), it is case sensitive. Would it be possible to get all entries starting with those letters, whatever the case?

2) Would it be possible to search inside other fields than bibtex keys, like author and title for example?

#### Example

For example, if I have the current entry in my `.bib` file:

```
@article{Key2017,
abstract = {abstract},
doi = {12345},
journal = {Nature},
month = {may},
pages = {54321},
title = {{This is an original title}},
volume = {8},
author = {firstauthor and author2},
year = {2017}
}
```
It would be great if we could type, for example:
- `\cite{k` (from bibtex key, case insensitive) 
- `\cite{ori` (from title) 
- `\cite{thi}` (also title) 
- `\cite{firsta` (from author) 

### Who will benefit with this feature?

Everybody!

**Comment by sunderme** [comment]
there are extra programs to manage and insert bibids. Maybe it is better to use those ?

**Comment by asetGem** [comment]
Do you mean programs like Mendeley or Zotero?

**Comment by sunderme** [comment]
yes, see also https://github.com/texstudio-org/texstudio-macro/blob/master/zotero-picker/zoteroPicker.txsMacro

**Comment by asetGem** [comment]
I am using Mendeley to manage my references and generate the `.bib`, but what I meant here was if texstudio could parse this `.bib` to allow autocompletion while writing the document.

**Comment by sunderme** [comment]
I understand. 
The completer GUI is principally limited in functionality, so coming from the Reference manager gives way better search capabilities.
